vllm/attention/backends/abstract.py

```diff
@@ -105,6 +107,11 @@ class AttentionMetadata:
     # in block 0, and 1st slot in block 1, respectively.
     slot_mapping: torch.Tensor
 
+    # The index maps that relate multi-modal embeddings to the corresponding
+    # placeholders.
+    multi_modal_placeholder_maps: Optional[Dict[
```

Comment:
I wish I could -- unfortunately because this is the base type for the other `AttentionMetadata` types, doing so would require that all fields in all derived types also have default values.

Reply:
And yeah I agree that the relationship with `AttentionMetadata` is sort of surprising -- the various attention implementations seem to own the projection from multiple sequences to flattened batch and the placeholder maps need to go along for that ride, but the audience for the maps themselves is obviously not the attention implementation.

So maybe a more pure (but somewhat bulkier) implementation would have `AttentionMetadataBuilder.build` return `tuple[AttentionMetadata, PlaceholderMap]` and the runners would be responsible for plumbing it all the way down to the model as a kwarg?

Reply:
I think having it inside the AttentionMetadata is okay for now, but could you update the comment here to reflect the fact that we make the maps part of the `AttentionMetadata` as a temporary solution to make it accessible to model_executable?